### PR TITLE
Add xeCJKVerbAddon to default fvset config for Chinese documents

### LINKED ISSUE #9414: CJK render issue in latexpdf code block
### Describe the bug

Some extra spaces are inserted in the code block if there are CJK characters.

### How to Reproduce

Create a simple project with 3 files.

```
==> conf.py <==
project = 'test'
copyright = '2021, test'
author = 'test'
language = 'zh_CN'
```
```
==> index.rst <==
.. code-block:: json

   {
     "test": "你好"
   }

.. code-block:: json

   {
     "test": "こんにちは"
   }
```
```
==> Makefile <==
# Minimal makefile for Sphinx documentation
#

# You can set these variables from the command line, and also
# from the environment for the first two.
SPHINXOPTS    ?=
SPHINXBUILD   ?= sphinx-build
SOURCEDIR     = .
BUILDDIR      = _build

# Put it first so that "make" without argument is like "make help".
help:
        @$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)

.PHONY: help Makefile

# Catch-all target: route all unknown targets to Sphinx using the new
# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
%: Makefile
        @$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```

Built with the docker

```
docker run --rm -it -v `pwd`:/docs sphinxdoc/sphinx-latexpdf
```

### Expected behavior

No extra spaces.

### Your project

none

### Screenshots

image

There are some extra space between `"` and the first CJK character.

### OS

Linux

### Python version

3.9.6

### Sphinx version

4.0.3

### Sphinx extensions

_No response_

### Extra tools

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zhsj**
### Feature or Bugfix

- Bugfix

### Purpose

Fixes: #9414

cc @stone-zeng @muzimuzhi

### CODE REVIEW

File: sphinx/builders/latex/constants.py [hunk 1]
```diff
@@ -198,6 +198,7 @@
         'polyglossia':  '',
         'babel':        '\\usepackage{babel}',
         'fontenc':      '\\usepackage{xeCJK}',
+        'fvset':        '\\fvset{fontsize=\\small,formatcom*=\\xeCJKVerbAddon}',
```
Comment (muzimuzhi) [inline sphinx/builders/latex/constants.py]: I made wrong assumptions in https://github.com/sphinx-doc/sphinx/issues/9414#issuecomment-875705288. 
 - When used in the optional key-val argument of `Verbatim`-derived environment, for example `\begin{Verbatim}[...]`, `formatcom*` is preferable. 
 - When used in `\fvset{...}` in (a well structured) preamble, `formatcom` is enough. And `formatcom*` is harmless.

Comment (zhsj) [reply sphinx/builders/latex/constants.py]: ok, i changed to `formatcom=`

**Comment by tk0miya** [comment]
Is this fix a temporary workaround or a configuration? I'd like to know we can remove this setting in the future or not.

Additionally, it would be better to add a comment on what this setting doing to maintain the code.

**Comment by zhsj** [comment]
> Is this fix a temporary workaround or a configuration? I'd like to know we can remove this setting in the future or not.

This is an old issue in xeCJK, which backs to 2017 (https://github.com/CTeX-org/ctex-kit/issues/331). It's not easy to fix on xeCJK, as fancyvrb is not in kernel(not sure the translation, I don't know what's that mean, but it's said in xeCJK issue, in Chinese.) Maybe @stone-zeng(xeCJK maintainer) can add some context here.

This is a configuration which proposed by @stone-zeng in https://github.com/sphinx-doc/sphinx/issues/9414#issuecomment-875676739

**Comment by zhsj** [comment]
> Additionally, it would be better to add a comment on what this setting doing to maintain the code.

done

**Comment by stone-zeng** [comment]
We may not patch `fancyvrb` packages etc in `xeCJK` recently, and even if the patch is added in the future, the settings here will still not be affected. So I think this pr is ok.

**Comment by tk0miya** [comment]
Thank you for explanation. Merging now.

**Pull request merged**